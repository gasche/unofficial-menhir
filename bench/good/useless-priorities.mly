%token <int> INT
%token PLUS TIMES
%left PLUS
%left TIMES
%token EOF
%start <int> main
%%
expression: i = INT { i } | e = expression; o = op; f = expression { o e f }
op: PLUS { ( + ) } | TIMES { ( * ) }
main: e = expression EOF { e }
