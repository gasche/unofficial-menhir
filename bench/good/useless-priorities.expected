%start main
%token <int> INT
%token EOF
%token PLUS
%token TIMES
%left PLUS 
%left TIMES 
%type <int> main
%%

expression:
| i = INT
    { i }
| e = expression o = op f = expression
    { o e f }

op:
| PLUS
    { ( + ) }
| TIMES
    { ( * ) }

main:
| e = expression EOF
    { e }

%%



